src/util.ts

```diff
@@ -882,3 +890,95 @@ export function getFormattedIpfsUrl(
   const cidAndPath = removeIpfsProtocolPrefix(ipfsUrl);
   return `${origin}/ipfs/${cidAndPath}`;
 }
+
+type PlainObject = Record<number | string | symbol, unknown>;
+
+/**
+ * Determines whether a value is a "plain" object.
+ *
+ * @param value - A value to check
+ * @returns True if the passed value is a plain object
+ */
+export function isPlainObject(value: unknown): value is PlainObject {
```

Comment:
Could you add unit tests for these new utility functions? They are all showing as covered by the permission controller unit tests, but they should be tested directly as well to ensure all uses cases/boundary conditions are considered.

Reply:
Done!